core/src/test/java/org/kiwix/kiwixmobile/core/utils/ExternalLinkOpenerTest.kt

```diff
@@ -34,48 +34,77 @@ import org.kiwix.kiwixmobile.core.extensions.toast
 
 internal class ExternalLinkOpenerTest {
   private val sharedPreferenceUtil: SharedPreferenceUtil = mockk()
-  private val alertDialogShower: AlertDialogShower = mockk()
+  private val alertDialogShower: AlertDialogShower = mockk(relaxed = true)
   private val intent: Intent = mockk()
   private val activity: Activity = mockk()
 
   @Test
-  internal fun `requestOpenLink is called if external link popup preference is true`() {
+  internal fun `alertDialogShower opens link if confirm-button is clicked`() {
     every { intent.resolveActivity(activity.packageManager) } returns mockk()
     every { sharedPreferenceUtil.prefExternalLinkPopup } returns true
+    val lambdaSlot = slot<() -> Unit>()
     val externalLinkOpener =
-      spyk(
-        ExternalLinkOpener(activity, sharedPreferenceUtil, alertDialogShower),
-        recordPrivateCalls = true
-      )
+      spyk(ExternalLinkOpener(activity, sharedPreferenceUtil, alertDialogShower))
     externalLinkOpener.openExternalUrl(intent)
-    verify { externalLinkOpener["requestOpenLink"](intent) }
+    verify {
+      alertDialogShower.show(any(), capture(lambdaSlot), any(), any())
+    }
+    lambdaSlot.captured.invoke()
+    verify { activity.startActivity(intent) }
   }
 
   @Test
-  internal fun `openLink is called if external link popup preference is false`() {
+  internal fun `alertDialogShower does not open link if negative-button is clicked`() {
+    every { intent.resolveActivity(activity.packageManager) } returns mockk()
+    every { sharedPreferenceUtil.prefExternalLinkPopup } returns true
+    val lambdaSlot = slot<() -> Unit>()
+    val externalLinkOpener =
+      spyk(ExternalLinkOpener(activity, sharedPreferenceUtil, alertDialogShower))
+    externalLinkOpener.openExternalUrl(intent)
+    verify {
+      alertDialogShower.show(any(), any(), capture(lambdaSlot), any())
+    }
+    lambdaSlot.captured.invoke()
+    verify(exactly = 0) { activity.startActivity(intent) }
+  }
+
+  @Test
+  internal fun `alertDialogShower opens link and saves preferences if neutral-button is clicked`() {
+    every { intent.resolveActivity(activity.packageManager) } returns mockk()
+    every { sharedPreferenceUtil.prefExternalLinkPopup } returns true
+    val lambdaSlot = slot<() -> Unit>()
+    val externalLinkOpener =
+      spyk(ExternalLinkOpener(activity, sharedPreferenceUtil, alertDialogShower))
+    externalLinkOpener.openExternalUrl(intent)
+    verify {
+      alertDialogShower.show(any(), any(), any(), capture(lambdaSlot))
```

Comment:
this dialogShower doesn't show `any` dialog, it shows a very specific dialog so I would match that parameter

Reply:
That's true!